shared/promptutil/prompt.go

```diff
@@ -99,3 +114,55 @@ func PasswordPrompt(promptText string, validateFunc func(string) error) (string,
 	}
 	return response, nil
 }
+
+// InputPassword with a custom validator along capabilities of confirming
+// the password and reading it from disk if a specified flag is set.
+func InputPassword(
+	cliCtx *cli.Context,
+	passwordFileFlag *cli.StringFlag,
+	promptText string,
+	confirmText string,
+	confirmPassword PasswordConfirm,
+	passwordValidator func(input string) error,
+) (string, error) {
+	if cliCtx.IsSet(passwordFileFlag.Name) {
+		passwordFilePathInput := cliCtx.String(passwordFileFlag.Name)
+		passwordFilePath, err := fileutil.ExpandPath(passwordFilePathInput)
+		if err != nil {
+			return "", errors.Wrap(err, "could not determine absolute path of password file")
+		}
+		data, err := ioutil.ReadFile(passwordFilePath)
+		if err != nil {
+			return "", errors.Wrap(err, "could not read password file")
+		}
+		enteredPassword := strings.TrimRight(string(data), "\r\n")
+		if err := passwordValidator(enteredPassword); err != nil {
+			return "", errors.Wrap(err, "password did not pass validation")
+		}
+		return enteredPassword, nil
+	}
+	var hasValidPassword bool
+	var password string
+	var err error
+	for !hasValidPassword {
+		password, err = PasswordPrompt(promptText, passwordValidator)
+		if err != nil {
+			return "", fmt.Errorf("could not read password: %v", err)
+		}
+
+		if confirmPassword == confirmPassword {
```

Comment:
`confirmPassword == confirmPassword` ? always gonna be true?

Reply:
Poor naming of variables, right hand side is an enum, left hand side is a user-specified input